src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+
+import java.io.*;
+
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+
+public class MovieRecommender {
+
+
+    private BiMap<Integer, String> mapProducts = HashBiMap.create();
+    private BiMap<Integer, String> mapUsers = HashBiMap.create();
+
+    List<String> usersIdArray = new ArrayList<>();
+    List<String> productsIdArray = new ArrayList<>();
+    List<Double> reviewsArray = new ArrayList<>();
+
+    private File fileInput;
+    private FileWriter fileWriter;
+
+    public MovieRecommender(String path) throws IOException {
+        this.fileWriter = new FileWriter("movies.csv");
+        this.readInputFile(path);
+    }
+
+    public int getTotalReviews(){
+        return this.reviewsArray.size();
+    }
+
+    public int getTotalProducts(){
+        return this.mapProducts.size();
+    }
+
+    public int getTotalUsers(){
+        return this.mapUsers.size();
+    }
+
+
+
+    private void readInputFile(String path) throws IOException {
+
+
+        String PRODUCT_ID ="product/productId: ";
+        String USER_ID = "review/userId: ";
+        String SCORE = "review/score: ";
+
+        String readedLine;
+
+        InputStream fileStream = null;
+        fileStream = new FileInputStream(path);
+        InputStream gzipStream = new GZIPInputStream(fileStream);
+        Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
+        BufferedReader bufferedReader = new BufferedReader(decoder);
+
+        int mapUserKey = 0;
+        int mapProductKey = 0;
+
+        while((readedLine = bufferedReader.readLine()) != null){
+            if(readedLine.contains(PRODUCT_ID)){
+                String splitedLine = readedLine.split(PRODUCT_ID)[1];
+                this.productsIdArray.add(splitedLine);
+                if(!this.mapProducts.containsValue(splitedLine)){
+                    this.mapProducts.put(mapProductKey, splitedLine);
+                    mapProductKey++;
+                }
+            }else if(readedLine.contains(USER_ID)){
+                String splitedLine = readedLine.split(USER_ID)[1];
+                this.usersIdArray.add(splitedLine);
+                if(!this.mapUsers.containsValue(splitedLine)){
+                    this.mapUsers.put(mapUserKey, splitedLine);
+                    mapUserKey++;
+                }
+            }else if(readedLine.contains(SCORE)){
+                String splitedLine = readedLine.split(SCORE)[1];
+                this.reviewsArray.add(Double.parseDouble(splitedLine));
+            }
+        }
+        this.writeCSV();
+        System.out.println("csv created");
+        System.out.println(this.mapProducts.size());
+        System.out.println(this.mapUsers.size());
+        System.out.println(this.reviewsArray.size());
+        this.closeCSV();
+    }
+
+    public BiMap<Integer, String> getMap(BiMap<Integer, String> biMap, String data){
```

Comment:
Are you using this function??

Reply:
:o, thanks for pointing that out, i'm not using it, i will delete it.